### PR TITLE
GUI: Allow dict update on custom frontend (#932)

### LINKED ISSUE #932: BUG-onChange callback not called for JSON update in gui
To reproduce

- Clone or pull from repositoy <https://github.com/bengaid/covid_taipy_chalkit>
- Install Chalk'it with Taipy link
```sh
pip install py-chalk-it-0.4.0.tar.gz
```
- Run the example
```sh
python covid_demo.py
```

When the user changes the country, the Python callback is correctly called, option_e dict is correctly computed. It is sent to front-end but the onChange callback is not called.

### PR DESCRIPTION
**Pull request by dinhlongviolin1**
Resolve #932

### CODE REVIEW

**Review by FabienLelaquais — DISMISSED** [review]
😶

File: taipy/gui/gui.py [hunk 1]
```diff
@@ -1002,7 +1002,9 @@ def __send_var_list_update(  # noqa C901
                 elif isinstance(newvalue, _TaipyToJson):
                     newvalue = newvalue.get()
                 if isinstance(newvalue, (dict, _MapDict)):
-                    continue  # this var has no transformer
+                    # Skip in taipy-gui, available in custom frontend
+                    if request.cookies.get(_Server._RESOURCE_HANDLER_ARG, None) is None:
```
Comment (FredLL-Avaiga) [inline taipy/gui/gui.py]: Are we sure the request is always available ?
what if the update is done from a long callback ?

Comment (FredLL-Avaiga) [reply taipy/gui/gui.py]: should we at least try/except this or check the availability or request and request.cookies ?

**Review by FredLL-Avaiga — DISMISSED** [review]

File: taipy/gui/gui.py (hunk 1 above)
Comment (dinhlongviolin1) [reply taipy/gui/gui.py]: I more than agree with you!

**Review by FredLL-Avaiga — DISMISSED** [review]
😍

**Review by FredLL-Avaiga — APPROVED** [review]
🎯

**Review by FabienLelaquais — APPROVED** [review]
Sure, why not

**Pull request merged**